clang/docs/checkedc/3C/INSTALL.md

```diff
@@ -0,0 +1,54 @@
+
+## Supported environments
+
+We work primarily on Linux.  Checked C supports Windows, but we do not test 3C regularly on Windows.  Some of us use Mac OS X, which Checked C does not officially support; we have gotten 3C to mostly work with some additional steps (see below), but there are some known problems and [Microsoft is still considering whether to work with us to address them](https://github.com/microsoft/checkedc/issues/424#issuecomment-720648442).
+
+## Basics
+
+**(TODO: Test from start to finish!)**
+
+Assuming you have already cloned the `checkedc-clang` repository, run the following (from the root of the working tree) for a basic build:
+
+```
+# See https://github.com/correctcomputation/checkedc-clang/issues/296 about this.
+(llvm/projects/checkedc-wrapper && git clone https://github.com/microsoft/checkedc)
+
+mkdir build && cd build
+cmake ../llvm -G Ninja -DLLVM_ENABLE_PROJECTS=clang
+ninja TARGET
+```
+
+where TARGET stands for the target you want to build.  For the `3c` command-line tool, the target is `3c`.  Executables will end up in `build/bin`.
+
+## Build options
+
+Because all of LLVM and clang are built as dependencies, this may consume up to ~50 GB of disk space, take several hours (depending on the speed of your computer), and require ~10 GB of RAM.  Incremental builds will usually be faster.  Here are some things you can do that may reduce the build time and/or peak memory use:
+
+- The above instructions already assume the use of the [Ninja](https://ninja-build.org/) build tool; you may have to install it.  You can alternatively use `make` (remove `-G Ninja` from the `cmake` command and replace `ninja` with `make`), but Ninja is much faster in our experience.
+- Pass `-DLLVM_TARGETS_TO_BUILD=X86` (or whatever your machine’s architecture is, e.g., `ARM`) to `cmake` to build only the parts of `clang``needed to compile programs for your machine.
+- Pass `-DLLVM_USE_LINKER=lld`.  This requires a sufficiently recent version of `lld` to be installed on your system.
+- Pass `-DLLVM_USE_SPLIT_DWARF=ON`.  We have tested the build with this option, but it may affect some debugging tools.
+- Pass `-DLLVM_OPTIMIZED_TABLEGEN=ON`.
+(The reference for improving build performance is [this LLVM page](https://www.llvm.org/docs/GettingStarted.html#common-problems), but we have sought to describe the most promising options here.)
+
+You might want to use `-DCMAKE_BUILD_TYPE=RelWithDebInfo` if you are running 3C enough between builds that the faster running time outweighs the slower build time or you simply want to test a release build of 3C, assuming that's what end users will eventually be encouraged to use.
+
+
+## Mac OS X
+
+(TODO: Review?)
```

Comment:
How so?

Reply:
Since I'm not familiar with this topic, I just copied the text from the old document.  I don't know if it needs further review.  If you're happy with it, I am.

Reply:
Yes it seems fine.